### PR TITLE
Handle error thrown by nestmate visibility check

### PR DESCRIPTION
**Pull request by fengxue-IS**
`checkVisibility` used to only return error code, Nestmates imposed
extra checks on the nesthost class which may throw errors.

This change addresses potential incorrect exception handling that
sets wrong type of exception during method lookup.

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
@gacholio can you please take a look?

File: runtime/vm/lookupmethod.c [hunk 1]
```diff
@@ -379,7 +379,9 @@ javaResolveInterfaceMethods(J9VMThread *currentThread, J9Class *targetClass, J9R
 			BOOLEAN shouldAdd = TRUE;
 			J9Class* newMethodClass = NULL;
 			newMethod = processMethod(currentThread, lookupOptions, foundMethod, interfaceClass, &data->exception, &data->exceptionClass, &data->errorType, nameAndSig, senderClass, targetClass);
-			if (NULL == newMethod) {
+			if (NULL != currentThread->currentException) {
+				return NULL;
```
Comment (gacholio) [inline runtime/vm/lookupmethod.c]: Can this one be a goto like the others rather than an inline return?

**Comment by gacholio** [comment]
Has this behaviour been verified against the RI?

File: runtime/vm/lookupmethod.c [hunk 2]
```diff
@@ -379,7 +379,9 @@ javaResolveInterfaceMethods(J9VMThread *currentThread, J9Class *targetClass, J9R
...
+			} else if (NULL == newMethod) {
```
Comment (gacholio) [inline runtime/vm/lookupmethod.c]: No real need for elseif if the previous returns or goto's.

File: runtime/vm/lookupmethod.c (hunk 1 above)
Comment (fengxue-IS) [reply runtime/vm/lookupmethod.c]: this function uses inline return consistently, would you like me to change all usage to `goto end`?

**Comment by fengxue-IS** [comment]
This fixed a JDK11 nestmates test failure to match RI behaviour

**Comment by gacholio** [comment]
jenkins test sanity zlinux jdk11

**Pull request merged**